### ISSUE TITLE
Learn from other mailboxes

### ISSUE THREAD

**Issue by neckhair** [carlostrub/sisyphus#9]
Hi and thanks for this great project! I have used it for about 3 weeks now and it already works better than my previous solution.

What do you think about adding an option to add more mailboxes to learn from? I am trying to keep my inbox tidy. This means I only keep mails I have to catch up on in my inbox. Other things go to the `.Archive` folder as soon as I have read them. This means Sisyphus cannot learn from these archived mails, unless I keep them in the inbox until Sisyphus had the chance to learn it.

The folder strucutre looks (simplified) like this:
```
|- cur
|- new
|- .Archive
   |- cur
   |- new
```
So what Sisyphus should do is to learn from either all of the `cur` folders in the mailbox directory (except `./.Junk/cur`). Or better, accept an option to configure which ones.

Like this I could keep my workflow and additionally Sisyphus could learn from all my 20'000 e-mails in the archive.

**Comment by carlostrub** [comment]
Thank you very much for your nice comment. I think that is a great idea. As I would really like to keep the 12FactorApp suggestion and have the configuration in environment variables, let me check how to best configure that. Maybe something like a comma-separated list of subdirectories?

My biggest concern is that currently, I am allowing sisyphus to run in parallel on several Maildirs. Not sure a user really wants to have the configuration the same in all the maildirs. Let me check with Viper's options to make sensible environment variable reads.

**Comment by neckhair** [comment]
Yeah, configuration is often a difficult topic. Combining the multiple Mailboxes approaches with the multiple folders won't be possible to solve cleanly. As far as I know Viper allows you to use environment variables and config files together. Maybe that could be an approach. But when it gets more complex a clean config file can be much easier to maintain.

```yaml
duration: 24h
mailboxes:
- path: /path/to/mailbox
  ham_folders:
    - ./
    - ./Archive
  spam_folder: .Junk
```
